osu.Game/Beatmaps/WorkingBeatmapCache.cs

```diff
@@ -129,12 +134,41 @@ private class BeatmapManagerWorkingBeatmap : WorkingBeatmap
             [NotNull]
             private readonly IBeatmapResourceProvider resources;
 
+            private IBeatmap cachedPlayableBeatmap;
+            private string cachedRulesetShortName;
+            private string cachedModsHash;
```

Comment:
The latest changes mean that playable beatmaps are now *cached local to the working beatmap instance that produced them*, which means that the claims of "improving performance when rapidly changing the selected map" are almost completely wiped out because song select spams `GetWorkingBeatmap()` like there's no tomorrow, completely bypassing the cache a level higher. In back-of-envelope 30-second testing this "cache" has a whopping hit rate of like *10%*.

I'm not sure which direction this PR is attempting to go in but so far I only see potential drawbacks that far outweigh a cache hit rate of 10% in the scenario that this PR is *explicitly attempting to optimise for*.